test/vms.changefeed.test.js

```diff
@@ -0,0 +1,584 @@
+        if (err) {
+            return callback(err);
+        }
+
+        return callback(null, checkEqual(body[key], value));
+    });
+}
+
+function waitForValue(url, key, value, callback) {
+
+    function onReady(err, ready) {
+        if (err) {
+            callback(err);
+            return;
+        }
+
+        if (!ready) {
+            times++;
+
+            if (times == TIMEOUT) {
+                throw new Error('Timeout waiting on ' + url);
+            } else {
+                setTimeout(function () {
+                    waitForValue(url, key, value, callback);
+                }, 1000);
+            }
+        } else {
+            times = 0;
+            callback(null);
+        }
+    }
+
+    return checkValue(url, key, value, onReady);
+}
+
+function waitForNicState(t, query, state, waitCallback) {
+    var stop = false;
+    var count = 0;
+    var maxSeconds = 60;
+
+    function getNicStatus(callback) {
+        client.napi.get({
+            path: '/nics',
+            query: query
+        }, function (err, req, res, nics) {
+            if (err) {
+                return callback(err);
+            } else if (!nics.length || !nics[0].state) {
+                // Log the state of the nics so that we know why we failed
+                t.deepEqual(nics, {}, 'nics - query: ' + JSON.stringify(query));
+                return callback(new Error('VM does not have valid NICs'));
+            } else {
+                return callback(null, nics[0].state);
+            }
+        });
+    }
+
+    async.doWhilst(
+        function (callback) {
+            getNicStatus(function (err, nicState) {
+                if (err) {
+                    return callback(err);
+                }
+
+                count++;
+                // Assume just one NIC
+                if (nicState === state) {
+                    stop = true;
+                    return callback();
+                } else if (count === maxSeconds) {
+                    stop = true;
+                    return callback(new Error('Timeout waiting on NIC state ' +
+                        'change from ' + nicState + ' to ' + state));
+                }
+
+                setTimeout(callback, 1000);
+            });
+        },
+        function () { return !stop; },
+        waitCallback);
+}
+
+function createOpts(path, params) {
+    return {
+        path: path,
+        headers: {
+            'x-request-id': uuid.create(),
+            'x-context': JSON.stringify({
+                caller: CALLER,
+                params: params || {}
+            })
+        }
+    };
+}
+
+exports.setUp = function (callback) {
+    common.setUp(function (err, _client) {
+        assert.ifError(err);
+        assert.ok(_client, 'restify client');
+        client = _client;
+        callback();
+    });
+};
+
+exports.find_headnode = function (t) {
+    client.cnapi.get('/servers', function (err, req, res, servers) {
```

Comment:
No reason... I just stole it from the vms.full.test.js file :). I can update it to use that though.